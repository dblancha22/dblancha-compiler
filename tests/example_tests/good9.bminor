// test comparotors and boolean logic
if ( a >= 5 )
{
	print "a >= 5";
}
if( a > 5 )
{
	print "a > 5";
}
if( a <= 5 )
{
	print "a <= 5";
}
if( a < 5 )
{
	print "a < 5";
}
if( a == 5 )
{
	print "a == 5";
}
if( a != 5 )
{
	print "a != 5";
}
a = 5; // test assigment
if (a || b)
{
	print " a || b is true";
}
if (a && b)
{
	print " a && b is true";
}
