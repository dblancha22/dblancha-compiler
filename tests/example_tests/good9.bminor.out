COMMENT // test comparotors and boolean logic
IF if
OPEN_PAREN (
IDENTIFIER a
GE >=
INTEGER_LITERAL 5
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL a >= 5
SEMICOLON ;
CLOSE_CURLY }
IF if
OPEN_PAREN (
IDENTIFIER a
GT >
INTEGER_LITERAL 5
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL a > 5
SEMICOLON ;
CLOSE_CURLY }
IF if
OPEN_PAREN (
IDENTIFIER a
LE <=
INTEGER_LITERAL 5
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL a <= 5
SEMICOLON ;
CLOSE_CURLY }
IF if
OPEN_PAREN (
IDENTIFIER a
LT <
INTEGER_LITERAL 5
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL a < 5
SEMICOLON ;
CLOSE_CURLY }
IF if
OPEN_PAREN (
IDENTIFIER a
EE ==
INTEGER_LITERAL 5
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL a == 5
SEMICOLON ;
CLOSE_CURLY }
IF if
OPEN_PAREN (
IDENTIFIER a
NE !=
INTEGER_LITERAL 5
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL a != 5
SEMICOLON ;
CLOSE_CURLY }
IDENTIFIER a
EQUALS =
INTEGER_LITERAL 5
SEMICOLON ;
COMMENT // test assigment
IF if
OPEN_PAREN (
IDENTIFIER a
OR ||
IDENTIFIER b
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL  a || b is true
SEMICOLON ;
CLOSE_CURLY }
IF if
OPEN_PAREN (
IDENTIFIER a
AND &&
IDENTIFIER b
CLOSE_PAREN )
OPEN_CURLY {
PRINT print
STRING_LITERAL  a && b is true
SEMICOLON ;
CLOSE_CURLY }
